lib/utils/eureka/src/index.js

```diff
@@ -5,33 +5,38 @@
 const request = require('abacus-request');
 const perf = require('abacus-perf');
 const urienv = require('abacus-urienv');
+const oauth = require('abacus-oauth');
 
 // Setup debug log
 const debug = require('abacus-debug')('abacus-eureka');
 
+// Return the Eureka v2 API URL. Can be configured with a EUREKA env variable
+// and default to http://localhost:8080.
+const uris = urienv({
+  eureka: 9990,
+  auth_server: 9882
+});
+
 // Return an Express middleware that responds to a health check request
 const health = () => {
   return function(req, res, next) {
-    if(req.path == '/healthcheck') {
-
-      // Get the health of the app computed from the perf stats
-      const h = perf.healthy();
-      debug('Returning app health %s', h);
-      res.status(h ? 200 : 500).send({
-        healthy: h
-      });
-    }
+    if(req.path == '/healthcheck')
+      oauth.basicAuth(uris.auth_server, req.headers &&
```

Comment:
The oauth.basicAuth function seems to bundle 3 unrelated capabilities:
- extracting the user and password from the Authorization header
- authenticating the user using UAA
- authorizing the user to perform something (here, perform a health check)

I think this should translate to 3 different functions.